contracts/PushComm/PushCommV3.sol

```diff
@@ -552,99 +563,118 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3, PausableUp
-    function createRequestWithFeeId(
-        CrossChainRequestTypes.ArbitraryRequestPayload memory _payload,
-        uint256 _amount,
-        uint256 _gasLimit
-    )
-        external
-        payable
-        whenNotPaused
-    {
-        require(_amount > 0, "Invalid Amount");
-        require(_payload.feePercentage <= 100, "Invalid Fee Percentage");
-
-        bytes memory arbitraryReqPayload = abi.encode(_payload);
-        bytes memory requestPayload = abi.encode(arbitraryReqPayload, msg.sender, CrossChainRequestTypes.RequestType.ArbitraryReq);
-        createCrossChainRequest(requestPayload, _amount, _gasLimit);
-    }
-
+    /**
+     * @notice Quotes the cost of bridging tokens to the recipient chain
+     * @dev Calls the Wormhole Transceiver to get the delivery price
+     * @return cost The cost of bridging tokens
+     */
     function quoteTokenBridgingCost() public view returns (uint256 cost) {
         TransceiverStructs.TransceiverInstruction memory transceiverInstruction =
             TransceiverStructs.TransceiverInstruction({ index: 0, payload: abi.encodePacked(false) });
         cost = WORMHOLE_TRANSCEIVER.quoteDeliveryPrice(WORMHOLE_RECIPIENT_CHAIN, transceiverInstruction);
     }
 
+    /**
+     * @notice Quotes the cost of relaying a message to the target chain with the specified gas limit
+     * @dev Calls the Wormhole Relayer to get the EVM delivery price
+     * @param targetChain The chain to which the message is being relayed
+     * @param gasLimit The gas limit for the message relay
+     * @return cost The cost of relaying the message
+     */
     function quoteMsgRelayCost(uint16 targetChain, uint256 gasLimit) public view returns (uint256 cost) {
         (cost,) = WORMHOLE_RELAYER.quoteEVMDeliveryPrice(targetChain, 0, gasLimit);
     }
 
-    function createCrossChainRequest(bytes memory _requestPayload, uint256 _amount, uint256 _gasLimit) public payable {
-        bytes32 recipient = BaseHelper.addressToBytes32(EPNSCoreAddress);
-        bytes32 sender = BaseHelper.addressToBytes32(msg.sender);
+    /**
+     * @notice Creates a cross-chain request based on the specified function type and payload
+     * @dev Implements restrictions and calls the internal function to create the cross-chain request
+     * @param functionType The type of cross-chain function to execute
+     * @param payload The payload data for the cross-chain request
+     * @param amount The amount of tokens to be transferred
+     * @param gasLimit The gas limit for the cross-chain request
+     */
+    function createCrossChain(
+        CrossChainRequestTypes.CrossChainFunction functionType,
+        bytes calldata payload,
+        uint256 amount,
+        uint256 gasLimit
+    ) external payable whenNotPaused {
+        // Implement restrictions based on functionType
+        if (functionType == CrossChainRequestTypes.CrossChainFunction.AddChannel) {
+            if (amount < ADD_CHANNEL_MIN_FEES) {
```

Comment:
generally if-revert consumes less gas than require, but will check

Reply:
checked it, changing to require increases the gas by ~2k units: [if-revert-Tx](https://sepolia.basescan.org/tx/0xd1962512caa00cc7b945440a7ca6315bb2c930d70d0e36e4e8cc2ba20d2fdbcb) , [require-Tx](https://sepolia.basescan.org/tx/0xf5e060b424810d80785ec80dd2609fb54732d1cba9129f7cdefd264a07776d68)

added this for the check
<img width="856" alt="image" src="https://github.com/push-protocol/push-smart-contracts/assets/54444545/4a53b614-0c41-4c00-9bbf-4315bb94731e">

since in the require, it evaluates both functionType and amount expressions
while in if-else, it only evaluates functionType check and moves on if not satisfied